operator/cmd/config/mock.go

```diff
@@ -0,0 +1,15 @@
+package config
+
+import (
+	"github.com/stretchr/testify/mock"
+	"k8s.io/client-go/rest"
+)
+
+type MockProvider struct {
```

Comment:
can we replace this mock implementation with an autogenerated one by `moq`? Just to be consistent with the other code base, and this way we would not have to implement tests for the mock